docs/06-contributors/999-rfcs/000-stories/story-16.md

```diff
@@ -0,0 +1,220 @@
+
+
+## Code 
+#### `tasklist.w`
+```ts (wing)
+bring cloud;
+bring redis;
+
+interface IMyRegExp {
+  inflight test(s: str): bool;
+}
+
+enum Status {
+  Uncompleted,
+  Completed
+}
+
+struct Task {
+  title: str;
+  estimation: duration?;
+  status: Status;
+}
+
+interface    {
+  inflight get(id: str): Task;
+  inflight add(task: Task): str;
+  inflight update(id: str, task: Task): str;
+  inflight remove(id: str): void; 
+  inflight find(predicate: inflight (Task): bool): Array<Task>;
+}
+
+resource TaskList implementes ITaskList {
+  _redis: redis.Redis;
+  // we are missing inflight init, so I need to create a lazy get_redis_client method  
+  // that creates the _redis_client when it is first called
+  inflight _redis_connection: redis.IRedisClient?; 
+  
+  extern "./tasklist_helper.js" static inflight uuid(): str; 
+  
+  init() {
+    this._redis = new redis.Redis();
+  }
+
+  inflight init() {
+    this._redis_connection = this._redis.ioredis();
+  }
+
+  inflight get(id: str): Task {
+     return Task.from_json(Json.parse(this._redis_connection.get(id)));
+  }
+  
+  inflight update(id: str, t: Task): str {
+    let s = Json.to_str(Task.to_json(t));
+    this._redis_connection.set(id, s);
+    this._redis_connection.sadd("todo", id);
+    return id;
+  } 
+  
+  inflight add(task: Task): str {
+    let id = TaskList.uuid();
+    print("adding task ${id} with data: ${task}"); 
+    return this.update(id, task);
+  }
+
+  inflight remove(id: str) {
+    print("removing task ${id}");
+    this._redis_connection.del(id);
+  }
+
+  inflight find(predicate: inflight (Task): bool ): Array<Task> { 
+    let result = MutArray<Task>[]; 
+    let ids = this._redis_connection.smembers("todo");
+    for id in ids {
+      let task = Task.from_json(Json.parse(this.get_redis_client.get(id)));
+      if predicate(task)  {
+        result.push(task);
+      }
+    }
+    return result.copy();
+  }
+
+}
+
+resource TaskListApi {
+  api: cloud.Api;
+  task_list: ITaskList;
+        
+  extern "./tasklist_helper.js" static inflight create_regex: (s: str): IMyRegExp  
+  extern "./tasklist_helper.js" static inflight get_data: (url: str): Json;
+        
+  init(task_list: ITaskList) {
+    this.task_list = task_list;
+    this.api = new cloud.Api();
+    
+    this.api.post("/tasks", inflight (req: cloud. Api.ApiRequest): cloud.ApiResponse => {
+      let var title = str.from_json(req.body.title);
+      // Easter Egg - if you add a todo with the single word "random" as the title, 
+      //              the system will fetch a random task from the internet
+      if title == "random" {
+        let data: Json = TaskListApi.get_data('https://www.boredapi.com/api/activity');
+        title = str.from_json(data.activity); 
+      } 
+      let id = this.task_list.add(Task {
+        title: title,
+        completed: Status.Uncompleted
+      });
+      return cloud.ApiResponse { status:201, body: Json.to_str(id) };
```

Comment:
`body` expects a `Json` so I think it should be
```suggestion
      return cloud.ApiResponse { status:201, body: Json.from_str(id) };
```
Or from our discussion the other day, we can implement a separate mechanism where `str` has implicit conversion to `Json` (not the same as subtyping)
```suggestion
      return cloud.ApiResponse { status:201, body: id };
```

Reply:
> (not the same as subtyping)
I appreciate your sensibility to my sensibilities 😆 

Are we fine with body being a json? Some API will expect a string